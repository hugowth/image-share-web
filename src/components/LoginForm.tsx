import React from "react";
import { Formik, Form, Field } from "formik";
import { Button, Box, Grid, TextField } from "@material-ui/core";
import { LoginPayload } from "../types";
import { object, string } from "yup";

type Props = {
  onSubmit: (values: LoginPayload) => Promise<void>;
};

const schema = object({
  username: string().required(),
  password: string().required(),
});

const LoginForm = ({ onSubmit }: Props) => {
  return (
    <div style={{ marginTop: 12 }}>
      <Formik
        initialValues={{ username: "", password: "" }}
        onSubmit={onSubmit}
        validationSchema={schema}
        validateOnMount
      >
        {(formik) => {
          const { isSubmitting, isValid, errors, touched } = formik;
          const disabledField = isSubmitting;
          const disabledSubmmit = !isValid || isSubmitting;

          return (
            <Form>
              <Grid container spacing={3}>
                <Grid item xs={12}>
                  <Field
                    as={TextField}
                    variant="outlined"
                    fullWidth
                    autoFocus
                    name="username"
                    label="User Name"
                    error={touched.username && Boolean(errors.username)}
                    helperText={errors.username}
                    disabled={disabledField}
                  />
                </Grid>
                <Grid item xs={12}>
                  <Field
                    as={TextField}
                    variant="outlined"
                    fullWidth
                    autoFocus
                    name="password"
                    label="Password"
                    error={touched.password && Boolean(errors.password)}
                    helperText={errors.password}
                    disabled={disabledField}
                  />
                </Grid>
              </Grid>

              <Box my={2}>
                <Button
                  type="submit"
                  variant="outlined"
                  color="primary"
                  disabled={disabledSubmmit}
                >
                  Login
                </Button>
              </Box>
            </Form>
          );
        }}
      </Formik>
    </div>
  );
};

export default LoginForm;
