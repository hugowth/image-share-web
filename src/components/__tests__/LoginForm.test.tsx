import React from "react";
import renderer from "react-test-renderer";

import LoginForm from "../LoginForm";

describe("LoginForm", () => {
  it("renders correctly", async () => {
    const onSubmit = jest.fn();
    const tree = renderer.create(<LoginForm onSubmit={onSubmit} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
