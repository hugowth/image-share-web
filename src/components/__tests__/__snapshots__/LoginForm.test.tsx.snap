// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<div
  style={
    Object {
      "marginTop": 12,
    }
  }
>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            User Name
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  User Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-5"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
      </button>
    </div>
  </form>
</div>
`;
